### PR TITLE
Adding `deadlink` from PyPI

### PR DESCRIPTION
**Pull request by sugatoray**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sugatoray** [comment]
### Build Error Reason

The strange thing is `appdirs` was already added to run-requirements! Still it failed. :exclamation: 

[(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=513372&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1011)

```sh
    import appdirs
ModuleNotFoundError: No module named 'appdirs'
```

**Comment by sugatoray** [comment]
### Build Error Reason

[(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=513378&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1019)

```sh
import httpx
ModuleNotFoundError: No module named 'httpx'
```

**Trial Fix**

Add `httpx` to host-requirements.

**Comment by sugatoray** [comment]
### Build Error Reason

```sh
import toml
ModuleNotFoundError: No module named 'toml'
```

**Trial Fix**

Add `toml` to host-requirements.

**Comment by sugatoray** [comment]
### Build Error Reason (4)

[(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=513383&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1103)

```sh
ModuleNotFoundError: No module named 'rich'
```

**Trial Fix**

Add `rich` to host-requirements.

**Comment by sugatoray** [comment]
### What if? :thought_balloon: 

Well, all the dependencies were added to host-requirements along with `flit-core` (imported as `flit_core`). This seems a bit strange. 

:bulb: What if flit-core is added to run-requirements instead of adding all the package dependencies to host-requirements?

**Comment by sugatoray** [comment]
PR is ready. Request for review/merging.

cc: @conda-forge/help-python

**Comment by sugatoray** [comment]
cc: @BastianZim

**Comment by BastianZim** [comment]
The deps are required because the package probably imports itself at some point of the build script which means that the run dependencies are required as well. It's an error upstream that you cannot really do anything about.

About the windows error, that's because you're not specifying an entry-point: https://github.com/nschloe/deadlink/blob/ed32f2a4443232a34f86df0a4d69eb09f9807434/pyproject.toml#L41

File: recipes/deadlink/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/deadlink-{{ version }}.tar.gz
+  sha256: a0fb4855994997b323f684e9022c2bef68980852ab1c4374ea0c4e508dd75e7d
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps
```
Comment (ndevenish) [inline recipes/deadlink/meta.yaml:16]: e.g.
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps
  entry_points:
    - deadlink = deadlink:cli
```
- on windows, the path rewriting that allows the console_scripts to work in linux/mac can't be done, so conda-build needs to be explicitly told about the entry_points

File: recipes/deadlink/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
```
Comment (BastianZim) [inline recipes/deadlink/meta.yaml:16]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv
```
This will be done by conda-forge.

**Review by BastianZim — CHANGES_REQUESTED** [review]

**Comment by sugatoray** [comment]
@BastianZim I believe that I have taken care of all the review-suggestions made so far. But I still see **"Changes requested"** in red. Am I missing something?

File: recipes/deadlink/meta.yaml (hunk 2 above)
Comment (sugatoray) [reply recipes/deadlink/meta.yaml]: Done.

**Comment by ndevenish** [comment]
I think that GitHub just sometimes doesn't realise, or that you might need to click "Resolved" on it. It's not a blocker. The "Awaiting Author contribution" label I think you can clear automatically by re-requesting review (as you did in https://github.com/conda-forge/staged-recipes/pull/19100#issuecomment-1140341691)

**Comment by BastianZim** [comment]
Yes, the label is removed by the bot. The Changes requested can only be removed by a maintainer, that is not done automatically and should be the same everywhere on GitHub.

**Comment by sugatoray** [comment]
Thank you for your review and help in merging the PR.

@BastianZim @ndevenish

**Pull request merged**